src/devices/cpu/unsp/unsp.cpp

```diff
@@ -299,7 +297,12 @@ void unsp_20_device::device_start()
 void unsp_device::device_reset()
 {
 	for (int i = 0; i < m_numregs; i++)
-		m_core->m_r[i] = 0;
+	{
+		if (i < m_numregs)
+			m_core->m_r[i] = 0;
+		else
+			m_core->m_r[i] = 0xdeadbeef;
+	}
```

Comment:
The `else` branch is unreachable here.  You need to change the loop condition to `i < ARRAY_LENGTH(m_numregs)` if you want to actually fill the unused entries in the array.

Reply:
ARRAY_LENGTH(m_core->m_r)  but yes, I'll do a new PR